keda/templates/12-keda-deployment.yaml

```
@@ -39,6 +39,7 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       serviceAccountName: {{ .Values.serviceAccount.name }}
+      automountServiceAccountToken: {{ .Values.automountServiceAccountToken }}
```

Let's use this approach instead:
```suggestion
      automountServiceAccountToken: {{ .Values.serviceAccount.automount.name }}
```

---

@tomkerkhove Thank you for suggested change. Unfortunately this will not work as autmountServiceAccountToken is a boolean field. Because of security restrictions we need to set autmountServiceAccountToken in serviceAccount to false and in deployments to true.

---

Let's go with this then?
```suggestion
      automountServiceAccountToken: {{ .Values.serviceAccount.automount.enabled }}
```

---

Oh I get it. So it's not related to the service account but rather the deployment instead?

What about this then?
```suggestion
      automountServiceAccountToken: {{ .Values.deployment.automount.enabled }}
```

---

That's OK for me, thanks! I'm not an expert in this area, does `automountServiceAccountToken` have requirements or involve any risk to break existing installations?

---

I have set serviceAccount and deployments to default behaviour of k8s so there is no risk for breaking existing installations. Additionally the needed opt-in for mounting API-token is added to both keda and metrics deployment. Please review. Thank you for your support.